cura/Settings/ContainerManager.py

```diff
@@ -29,21 +29,115 @@ def __init__(self, parent = None):
     #   \return The ID of the new container, or an empty string if duplication failed.
     @pyqtSlot(str, result = str)
     def duplicateContainer(self, container_id):
-        containers = self._registry.findInstanceContainers(id = container_id)
+        containers = self._registry.findContainers(None, id = container_id)
         if not containers:
-            return
+            return ""
+
+        container = containers[0]
+
+        new_container = None
+        new_name = self._registry.uniqueName(container.getName())
+        if hasattr(container, "duplicate"):
+            new_container = container.duplicate(new_name)
+        else:
+            new_container = container.__class__(new_name)
+            new_container.deserialize(container.serialize())
+            new_container.setName(new_name)
+
+        if new_container:
+            self._registry.addContainer(new_container)
 
-        new_name = self_registry.uniqueName(containers[0].getName())
-        new_material = containers[0].duplicate(new_name)
-        self._registry.addContainer(new_material)
+        return new_container.getId()
 
-    @pyqtSlot(str, str)
-    def renameContainer(self, container_id, new_name):
-        pass
+    ##  Change the name of a specified container to a new name.
+    #
+    #   \param container_id \type{str} The ID of the container to change the name of.
+    #   \param new_id \type{str} The new ID of the container.
+    #   \param new_name \type{str} The new name of the specified container.
+    #
+    #   \return True if successful, False if not.
+    @pyqtSlot(str, str, str, result = bool)
+    def renameContainer(self, container_id, new_id, new_name):
```

Comment:
Name and ID can't be forced to be equal to each other. Otherwise we'll get a lot of name clashes if we try to make a Normal Quality profile for ABS 0.4mm and one for ABS 0.8mm and one for PLA 0.4mm, etc...

ID needs to be unique, and name is user-visible. That's what I held so far.